Python/bytecodes.c

```diff
@@ -3218,7 +3218,6 @@ dummy_func(
             assert(func->func_defaults);
             assert(PyTuple_CheckExact(func->func_defaults));
             int defcount = (int)PyTuple_GET_SIZE(func->func_defaults);
-            assert(defcount <= code->co_argcount);
```

Comment:
Please leave this. The assertion is correct, it is the optimization that is (currently) faulty.